Bottom: 0248c35a9bb009e90b0833384e4f4edb633f5a98
Top:    3ca44dba32af951288cd3165f1b7c3f24794bea5
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-05 16:17:30 +0800

Refresh of virtio-net-debugfs-for

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 1526014..a403443 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -221,6 +221,8 @@ struct virtnet_info {
 
 	/* failover when STANDBY feature enabled */
 	struct failover *failover;
+
+	struct dentry *ddir;
 };
 
 struct padded_vnet_hdr {
@@ -2370,7 +2372,25 @@ static int virtnet_get_phys_port_name(struct net_device *dev, char *buf,
 	return 0;
 }
 
+static int virtnet_init(struct net_device *dev)
+{
+	struct virtnet_info *vi = netdev_priv(dev);
+
+	vi->ddir = debugfs_create_dir(netdev_name(dev), virtnet_ddir);
+	if (IS_ERR_OR_NULL(vi->ddir))
+		return -ENOMEM;
+
+	return 0;
+}
+
+static void virtnet_uninit(struct net_device *dev)
+{
+	return;
+}
+
 static const struct net_device_ops virtnet_netdev = {
+	.ndo_init            = virtnet_init,
+	.ndo_uninit          = virtnet_uninit,
 	.ndo_open            = virtnet_open,
 	.ndo_stop   	     = virtnet_close,
 	.ndo_start_xmit      = start_xmit,
